homeassistant/components/trafikverket_ferry/sensor.py

```diff
@@ -0,0 +1,166 @@
+    if WEEKDAYS[today_weekday] in departure:
+        return today_date
+    for day in departure:
+        next_departure = WEEKDAYS.index(day)
+        if next_departure > today_weekday:
+            return next_weekday(today_date, next_departure)
+    return next_weekday(today_date, WEEKDAYS.index(departure[0]))
+
+
+def _to_iso_format(traintime: datetime) -> str:
+    """Return isoformatted utc time."""
+    return as_utc(traintime.replace(tzinfo=UTC)).isoformat()
+
+
+class FerrySensor(SensorEntity):
+    """Contains data about a train depature."""
+
+    _attr_icon = ICON
+    _attr_device_class = SensorDeviceClass.TIMESTAMP
+
+    def __init__(
+        self,
+        ferry_api: TrafikverketFerry,
+        name: str,
+        ferry_from: str,
+        ferry_to: str,
+        weekday: list,
+        departuretime: time | None,
+        entry_id: str,
+    ) -> None:
+        """Initialize the sensor."""
+        self._ferry_api = ferry_api
+        self._attr_name = name
+        self._ferry_from = ferry_from
+        self._ferry_to = ferry_to
+        self._weekday = weekday
+        self._time = departuretime
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, entry_id)},
+            manufacturer="Trafikverket",
+            model="v1.2",
+            name=name,
+            configuration_url="https://api.trafikinfo.trafikverket.se/",
+        )
+        self._attr_unique_id = create_unique_id(
+            ferry_from, ferry_to, departuretime, weekday
+        )
+
+    async def async_update(self) -> None:
+        """Retrieve latest state."""
+        when = datetime.now()
+        _state: FerryStop | None = None
+        if self._time:
+            departure_day = next_departuredate(self._weekday)
+            when = datetime.combine(departure_day, self._time)
+        try:
+            if self._time:
+                _state = await self._ferry_api.async_get_next_ferry_stop(
+                    self._ferry_from, self._ferry_to, when
+                )
+            else:
+
+                _state = await self._ferry_api.async_get_next_ferry_stop(
+                    self._ferry_from, self._ferry_to, when
+                )
+        except ValueError as output_error:
+            _LOGGER.error("Departure %s encountered a problem: %s", when, output_error)
+
+        if not _state:
+            self._attr_available = False
+            self._attr_native_value = None
+            self._attr_extra_state_attributes = {}
+            return
+
+        self._attr_available = True
+
+        self._attr_native_value = _state.departure_time.replace(tzinfo=UTC)
+
+        self._attr_extra_state_attributes = {
+            ATTR_FROM: _state.from_harbor_name,
+            ATTR_TO: _state.to_harbor_name,
+            ATTR_MODIFIED_TIME: _to_iso_format(_state.modified_time),
+            ATTR_OTHER_INFO: _state.other_information,
+        }
```

Comment:
It looks like these should be their own sensors

Reply:
I kept other info as attribute but the rest is now own sensors.
Other info is an unstructured list from the agency and would be pretty strange as a sensor.